esmvalcore/preprocessor/_area.py

```diff
@@ -56,20 +57,24 @@ def extract_region(cube, start_longitude, end_longitude, start_latitude,
     start_latitude = float(start_latitude)
     end_latitude = float(end_latitude)
 
+    # Regular grid
     if cube.coord('latitude').ndim == 1:
         region_subset = cube.intersection(
             longitude=(start_longitude, end_longitude),
             latitude=(start_latitude, end_latitude))
         region_subset = region_subset.intersection(longitude=(0., 360.))
         return region_subset
-    # irregular grids
+
+    # Irregular grids - not lazy.
     lats = cube.coord('latitude').points
     lons = cube.coord('longitude').points
-    select_lats = start_latitude < lats < end_latitude
-    select_lons = start_longitude < lons < end_longitude
-    selection = select_lats & select_lons
-    data = da.ma.masked_where(~selection, cube.core_data())
-    return cube.copy(data)
+    mask = np.ma.array(cube.data).mask
+    mask += np.ma.masked_where(lats < start_latitude, lats).mask
+    mask += np.ma.masked_where(lats > end_latitude, lats).mask
+    mask += np.ma.masked_where(lons < start_longitude, lons).mask
+    mask += np.ma.masked_where(lons > end_longitude, lons).mask
+    cube.data = da.ma.masked_array(data=cube.data, mask=mask)
```

Comment:
This doesn't look right:

- `masked_where(condition, array).mask` should be the same as `condition`
- On boolean arrays `+` is logical or  (and would be better written as |)

The resulting mask is probably correct, but the way is overly complicated and unnecessarily excludes dask.
I'll see if I can get something akin to the old code working correctly with both dask and numpy. @ledm, what exactly was the problem with the old code?

Reply:
```
select_lats = start_latitude < lats < end_latitude
```
is not a valid dask command and doesn't work. We should replace these lines with something compatible with dask. I just re-added the old numpy lines because they worked.

Reply:
Then this seems to be a bug in `extract_region` that has nothing to do with this fx file stuff, correct?
I suggest to

- Undo the changes here
- Open a `bug` issue
- Write a test that shows the problem
- Write a proper fix, maybe the more explicit `start_latitude < lats & lats < end_latitude` works.

We should not mix up completely independent things, less when the PR is as complicated as this one.

Reply:
PS: I am happy to help. Just let me know how you want to proceed and what part you would like to do yourself :)